monai/utils/type_conversion.py

```diff
@@ -199,6 +245,11 @@ def convert_data_type(
             data = convert_to_numpy(data)
         if data is not None and dtype != data.dtype:
             data = data.astype(dtype)
+    elif has_cp and output_type is cp.ndarray:
+        if orig_type is not cp.ndarray:
```

Comment:
Seems the `convert_to_cupy` API can support `cupy arrary` input, so you don't need to check it here again.
Just:
```py
elif has_cp and output_type is cp.ndarray:
    if data is not None:
        data = convert_to_cupy(data, dtype)
```
Right?

Reply:
Good observation but similar to numpy, in cupy, `astype` will make a copy by default so I don't want to `astype` being called if the type and dtype of input and output are the same. https://numpy.org/doc/stable/reference/generated/numpy.ndarray.astype.html#numpy.ndarray.astype

Reply:
I think maybe you mixed `asarray()` and `astype()` here? You can see the doc-string of `asarray()`: "If a is already on the device, no copy is performed."
https://docs.cupy.dev/en/stable/reference/generated/cupy.asarray.html

Thanks.

Reply:
Right, I have `astype` in the code but I can change it to `asarray` thanks.